src/wp-admin/includes/class-wp-screen.php

```diff
@@ -949,8 +951,9 @@ public function render_screen_meta() {
 		if ( $this->get_option( 'layout_columns' ) ) {
 			$this->columns = (int) get_user_option( "screen_layout_$this->id" );
 
-			if ( ! $this->columns && $this->get_option( 'layout_columns', 'default' ) ) {
-				$this->columns = $this->get_option( 'layout_columns', 'default' );
+			$layout_columns = (int) $this->get_option( 'layout_columns', 'default' );
+			if ( ! $this->columns && $layout_columns ) {
+				$this->columns = $layout_columns;
 			}
```

Comment:
Note that this code only runs in the classic edit post screen, as far as I can tell. I could only get it to run with the Classic Editor plugin active.